src/coreclr/pal/src/thread/thread.cpp

```diff
@@ -1676,12 +1688,28 @@ CorUnix::InternalSetThreadDescription(
 
     // Null terminate early.
     // pthread_setname_np only accepts up to 16 chars.
-    if (nameSize > 15)
+    #if defined(__linux__)
+    maxNameSize = 15;
+    #else
```

Comment:
This should be consistent with the rest of the file...

Reply:
Maybe we can define it as a macro outside of the function body (similar to macOS)?
```c
#ifdef __APPLE__
  MAX_THREAD_NAME_SIZE 63
#else
  MAX_THREAD_NAME_SIZE 15
#endif
```

Reply:
We do not need to have a magic number and truncate the string on Apple at all. We can just leave it for the API to do it for us.

Reply:
I _actually_ tested this at it looks like MacOS won't set the thread name if it's bigger than the maximum so I went with @am11's suggestion.

Reply:
True, just tested it myself and the resource string of error on macOS is `File name too long`.